Parent:     689305c5 (Remove AndroidRunner::asyncStop; always stop synchronously.)
Author:     Ray Donnelly <mingw.android@gmail.com>
AuthorDate: 2012-12-10 23:43:21 +0000
Commit:     Ray Donnelly <mingw.android@gmail.com>
CommitDate: 2012-12-13 12:38:48 +0000

Use QProcess::kill() rather than QProcess::terminate() for adb.exe

terminate (on Windows) requires a Windows Event Loop to process it
so use kill() on all platforms instead.

Change-Id: I39802bf228e14e52594fd359994668c7df8ffbd9
diff --git a/src/plugins/android/androidconfigurations.cpp b/src/plugins/android/androidconfigurations.cpp
index 8f2bbe0..23b23e9 100644
--- a/src/plugins/android/androidconfigurations.cpp
+++ b/src/plugins/android/androidconfigurations.cpp
@@ -411,7 +411,7 @@
     QProcess adbProc;
     adbProc.start(adbToolPath().toString(), QStringList() << QLatin1String("devices"));
     if (!adbProc.waitForFinished(-1)) {
-        adbProc.terminate();
+        adbProc.kill();
         return devices;
     }
     QList<QByteArray> adbDevs = adbProc.readAll().trimmed().split('\n');
@@ -564,7 +564,7 @@
     proc.start(adbToolPath().toString(), QStringList() << QLatin1String("-e") << QLatin1String("wait-for-device"));
     while (!proc.waitForFinished(500)) {
         if (avdProcess->waitForFinished(0)) {
-            proc.terminate();
+            proc.kill();
             proc.waitForFinished(-1);
             return QString();
         }
@@ -574,7 +574,7 @@
     // workaround for stupid adb bug
     proc.start(adbToolPath().toString(), QStringList() << QLatin1String("devices"));
     if (!proc.waitForFinished(-1)) {
-        proc.terminate();
+        proc.kill();
         return QString();
     }
 
@@ -596,7 +596,7 @@
                   << QLatin1String("shell") << QLatin1String("getprop")
                   << QLatin1String("ro.build.version.sdk"));
     if (!adbProc.waitForFinished(-1)) {
-        adbProc.terminate();
+        adbProc.kill();
         return -1;
     }
     return adbProc.readAll().trimmed().toInt();
diff --git a/src/plugins/android/androidmanager.cpp b/src/plugins/android/androidmanager.cpp
index e20af13..8ce5abe 100644
--- a/src/plugins/android/androidmanager.cpp
+++ b/src/plugins/android/androidmanager.cpp
@@ -856,7 +856,7 @@
     readelfProc.start(readelfPath.toString(), QStringList() << QLatin1String("-d") << QLatin1String("-W") << lib);
 
     if (!readelfProc.waitForFinished(-1)) {
-        readelfProc.terminate();
+        readelfProc.kill();
         return libs;
     }
 
diff --git a/src/plugins/android/androidpackagecreationstep.cpp b/src/plugins/android/androidpackagecreationstep.cpp
index 36a032f..a14873e 100644
--- a/src/plugins/android/androidpackagecreationstep.cpp
+++ b/src/plugins/android/androidpackagecreationstep.cpp
@@ -203,7 +203,7 @@
     readelfProc.start(AndroidConfigurations::instance().readelfPath(target()->activeRunConfiguration()->abi().architecture()).toString(),
                       QStringList() << QLatin1String("-d") << QLatin1String("-W") << appPath);
     if (!readelfProc.waitForFinished(-1)) {
-        readelfProc.terminate();
+        readelfProc.kill();
         return;
     }
     QStringList libs;
@@ -250,7 +250,7 @@
     }
     readelfProc.start(m_readElf.toString(), QStringList() << QLatin1String("-d") << QLatin1String("-W") << m_appPath.toUserOutput());
     if (!readelfProc.waitForFinished(-1)) {
-        readelfProc.terminate();
+        readelfProc.kill();
         return;
     }
     QStringList libs;
@@ -467,7 +467,7 @@
                            QStringList()<<QLatin1String("--strip-unneeded") << file);
         stripProcess.waitForStarted();
         if (!stripProcess.waitForFinished())
-            stripProcess.terminate();
+            stripProcess.kill();
     }
 }
 
diff --git a/src/plugins/android/androidrunner.cpp b/src/plugins/android/androidrunner.cpp
index a9fa2f9..69e1b7e 100644
--- a/src/plugins/android/androidrunner.cpp
+++ b/src/plugins/android/androidrunner.cpp
@@ -82,7 +82,7 @@
                  QStringList() << QLatin1String("-s") << m_deviceSerialNumber
                  << QLatin1String("shell") << QLatin1String("readlink") << QLatin1String("$(which ps)"));
     if (!psProc.waitForFinished(-1)) {
-        psProc.terminate();
+        psProc.kill();
         return;
     }
     QByteArray which = psProc.readAll();
@@ -251,7 +251,7 @@
 void AndroidRunner::stop()
 {
     QMutexLocker locker(&m_mutex);
-    m_adbLogcatProcess.terminate();
+    m_adbLogcatProcess.kill();
     m_adbLogcatProcess.waitForFinished(-1);
     m_checkPIDTimer.stop();
     if (m_processPID == -1)
@@ -301,7 +301,7 @@
 
     process.start(AndroidConfigurations::instance().adbToolPath().toString(), arguments);
     if (!process.waitForFinished(timeout))
-        process.terminate();
+        process.kill();
 }
 
 QString AndroidRunner::displayName() const
